playout/libretime_playout/liquidsoap/1.4/ls_script.liq

```diff
@@ -197,6 +197,15 @@ else
     s
 end
 
+def start_recording(arguments)
+    #E.g. {"filename" : "/app/recorder/test2.ogg", "length" : "25"}
+    args = of_json(default= [("error","fail")],arguments)
+    log("sources.start_recording #{args}")
+    recording_length = float_of_string(default=0.,args["length"])
+    recording = max_duration(recording_length,s)
+    output.file(%vorbis, args["filename"],fallible=true, recording)
+end
```

Comment:
This does not take into account the config recorder settings:
```
    record_file_format: Literal["mp3", "ogg"] = "ogg"  # record_file_type
    record_bitrate: int = 256
    record_samplerate: int = 44100
    record_channels: int = 2
    record_sample_size: int = 16
```

Reply:
It turns out that you can't use a variable as a parameter in Liquidsoap's encoding format (see https://www.liquidsoap.info/doc-1.4.4/encoding_formats.html) . 

This might have been fixed in 2.1.1. 

I've updated the code so that it uses the file format (mp3 or ogg).   But not sure what to do with the rest.  Perhaps leave the Liquidsoap defaults or offer a selection of defaults?